### ISSUE TITLE
Fine-tuning to get updated sentence embeddings

### ISSUE THREAD

**Issue by charlesxjyang** [UKPLab/sentence-transformers#190]
I have a corpus of text that I would like to fine-tune a model on. The goal is to get better embeddings based on the specific text corpus I have. I don't have any kind of labels for the corpus, they're just sentences. Is there any easy loss function or way to do this?

**Comment by nreimers** [comment]
Hi @charlesxjyang 
Sadly that is not possible. You need some notion of similarity that indicates whether two sentences are similar or not. Without that, it is not really to fine-tune it.

Sometimes you can derive this information from you text alone. See the IBM wikipedia dataset I mention in the paper. There, they use the information if two sentences are in the same paragraph / section or not.

Best
Nils Reimers

**Comment by kalyanks0611** [comment]
@charlesxjyang 
Initially,  you can further pre-train BERT or RoBERTa with the specific text corpusas suggested by [1] and then train this updated model using ALLNLI followed by STS.  This may improve the results at least to some extent.

[1] Don’t Stop Pretraining: Adapt Language Models to Domains and Tasks

What is your comment on this?(@nreimers ) Correct me, if I'm wrong.

**Comment by prince14322** [comment]
What kind of similarity notion is needed?
[sentence1, sentence2, X]
X -> 0 or 1 (0 : not similar and 1 : similar)
X -> cosine similarity ( real values between 0 and 1)

what else X could be?

**Comment by nreimers** [comment]
@prince14322 
These are the two common options.

Another option is triplet loss, where you have (anchor, positive, negative)
